File: testdata/langs/swift/Sample.swift
import Foundation
import Combine

protocol Repository {
    associatedtype Entity
    func findById(_ id: UUID) async throws -> Entity?
    func findAll() async throws -> [Entity]
    func save(_ entity: Entity) async throws -> Entity
    func delete(_ id: UUID) async throws
}

struct User: Codable, Identifiable {
    let id: UUID
    var name: String
    var email: String
}

enum UserRole: String, CaseIterable {
    case admin = "admin"
    case editor = "editor"
    case viewer = "viewer"
}

typealias UserCompletion = (Result<User, Error>) -> Void

class UserRepository: Repository {
    typealias Entity = User

    private var store: [User] = []

    func findById(_ id: UUID) async throws -> User? {
        store.first { $0.id == id }
    }

    func findAll() async throws -> [User] {
        store
    }

    func save(_ entity: User) async throws -> User {
        store.removeAll { $0.id == entity.id }
        store.append(entity)
        return entity
    }

    func delete(_ id: UUID) async throws {
        store.removeAll { $0.id == id }
    }
}

func createUser(name: String, email: String) async throws -> User {
    let user = User(id: UUID(), name: name, email: email)
    return try await UserRepository().save(user)
}

File: testdata/langs/kotlin/Sample.kt
package com.example.sample

import java.util.UUID
import kotlin.collections.List

interface Repository<T> {
    fun findById(id: UUID): T?
    fun findAll(): List<T>
    fun save(entity: T): T
    fun delete(id: UUID)
}

data class User(
    val id: UUID,
    val name: String,
    val email: String
)

enum class UserRole {
    ADMIN,
    EDITOR,
    VIEWER
}

object UserCache {
    private val cache = mutableMapOf<UUID, User>()

    fun get(id: UUID): User? = cache[id]
    fun put(user: User) {
        cache[user.id] = user
    }
}

class UserRepository : Repository<User> {
    private val store = mutableListOf<User>()

    override fun findById(id: UUID): User? =
        store.find { it.id == id }

    override fun findAll(): List<User> = store.toList()

    override fun save(entity: User): User {
        store.removeIf { it.id == entity.id }
        store.add(entity)
        UserCache.put(entity)
        return entity
    }

    override fun delete(id: UUID) {
        store.removeIf { it.id == id }
    }
}

fun createUser(name: String, email: String): User {
    val user = User(UUID.randomUUID(), name, email)
    return UserRepository().save(user)
}
